test/RoleHelperLib.sol

```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.8.6;
+pragma solidity >=0.5.0;
```

Comment:
This version constraint is unsafe here IMO because we use arithmetics (which had breaking changes in 0.8) - see lines 32, 65-67
So it should at least be `>= 0.8.0` IMO

Reply:
I actually thought about this and came to the conclusion that it would not be unsafe since this is not a deployed contract that would be attacked, and the only realistic scenario would be an overflow which would simply read a bad storage slot rather than reverting. But I overall still agree with you since there's really no reason anyone should be running this with <0.8 at this point.